target-gen/src/parser.rs

```diff
@@ -20,21 +23,15 @@ pub(crate) fn read_elf_bin_data<'a>(
         log::debug!("Segment address: {:#010x}", segment_address);
         log::debug!("Segment size:    {} bytes", segment_size);
 
-        // If the requested data is above the current segment, skip the segment.
-        if address > segment_address + segment_size {
-            continue;
-        }
-
-        // If the requested data is below the current segment, skip the segment.
-        if address + size <= segment_address {
+        let segment = segment_address..segment_address + segment_size;
+        // If the requested data is not fully inside of the current segment, skip the segment.
+        if !segment.contains(&range_to_read.start) || !segment.contains(&range_to_read.end) {
```

Comment:
Is this correct with inclusiveness? :) I think we even have a full contains function for ranges, but in the main lib afaik.

Reply:
Umm... it is, of course... not!